warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgePreconditionTest.java

```diff
@@ -478,6 +480,40 @@ public void testAwarePreconDifferentGroup() {
 
     }
 
+    @Test
+    public void testCategoryAwarePreconDifferentGroup() {
+        // CANINE == 'shepherd'
+
+        fields.put("EVENT_DATE", new BaseNormalizedContent("EVENT_DATE", "2022-10-26T01:31:53Z"));
+        fields.put("UUID", new BaseNormalizedContent("UUID", "0016dd72-0000-827d-dd4d-001b2163ba09"));
+        fields.put("ACTIVITY", new NormalizedFieldAndValue("ACTIVITY", "fetch", "THING", "0"));
+        fields.put("ACTIVITY", new NormalizedFieldAndValue("ACTIVITY", "swim", "THING", "1"));
+
+        LcNoDiacriticsListType listType = new LcNoDiacriticsListType();
+        NormalizedFieldAndValue normalizedContent = new NormalizedFieldAndValue("CANINE", "shepherd,bernese,ketchup", "PET", "0");
+        NormalizedContentInterface copy = new NormalizedFieldAndValue(normalizedContent);
+
+        // test edge config only allows FULL category. LIST_ELEMENTS edges should not be produced in this case
+        for (Pair<String,datawave.data.type.Type.Category> one : listType.normalizeToMany(copy.getIndexedFieldValue())) {
+            copy.setIndexedFieldValue(one.getFirst());
+            copy.setTypeCategory(one.getSecond());
+            fields.put("CANINE", copy);
+            copy = new NormalizedFieldAndValue(normalizedContent);
+        }
+
+        ProtobufEdgeDataTypeHandler<Text,BulkIngestKey,Value> edgeHandler = new ProtobufEdgeDataTypeHandler<>();
+        TaskAttemptContext context = new TaskAttemptContextImpl(conf, new TaskAttemptID());
+        edgeHandler.setup(context);
+
+        Set<String> expectedKeys = new HashSet<>();
+
+        RawRecordContainer myEvent = getEvent(conf);
+
+        EdgeHandlerTestUtil.processEvent(fields, edgeHandler, myEvent, 0, true, false);
+        Assert.assertEquals(expectedKeys, EdgeHandlerTestUtil.edgeKeyResults.keySet());
```

Comment:
If this test is demonstrating that no edge keys are produced I would rename the method and rewrite the assertions to be clear. Someone who isn't familiar with this code like myself might wonder what edges are produced when the preconditions belong to different groups, see that we create a set of expected keys, and then go looking for where we updated the expected collection.

Reply:
"// test edge config only allows FULL category. LIST_ELEMENTS edges should not be produced in this case"

No problem renaming the test.  When I hit these in other MRs I usually expand the rest of the Test class to see what is unaffected and other cases currently handled.

"When the preconditions belong to different groups" ?  Not sure what you mean there.  You have one precondition per edge, each edge is aware or unaware of the group, and the fields belong to different groups or the same group. https://github.com/NationalSecurityAgency/datawave/blob/320f29cb95c3c4e70fa8c8131b0dd2c9230f16c5/warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgePreconditionTest.java#L313